yt_dlp/extractor/trtcocuk.py

```diff
@@ -0,0 +1,47 @@
+from urllib.parse import unquote
+
+from .common import InfoExtractor
+from ..utils import int_or_none, parse_iso8601, traverse_obj
+
+
+class TrtCocukVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://www\.trtcocuk\.net\.tr/video/(?P<id>[\w-]+)'
+    _TESTS = [{
+        'url': 'https://www.trtcocuk.net.tr/video/kaptan-pengu-ve-arkadaslari-1',
+        'info_dict': {
+            'id': '3789738',
+            'ext': 'mp4',
+            'season_number': 1,
+            'series': '"Kaptan Pengu ve Arkadaşları"',
+            'season': 'Season 1',
+            'title': 'Kaptan Pengu ve Arkadaşları 1 Bölüm İzle TRT Çocuk',
+            'release_date': '20201209',
+            'release_timestamp': 1607513774,
+        }
+    }, {
+        'url': 'https://www.trtcocuk.net.tr/video/sef-rokanin-lezzet-dunyasi-17',
+        'info_dict': {
+            'id': '10260842',
+            'ext': 'mp4',
+            'series': '"Şef Roka\'nın Lezzet Dünyası"',
+            'title': 'Şef Roka\'nın Lezzet Dünyası 17 Bölüm İzle TRT Çocuk',
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        nuxtjs_data = self._search_nuxt_data(webpage, display_id)['data']
+        video_url = self._search_regex(
+            r'[\'"]?(?P<url>[^"]+)', unquote(nuxtjs_data['video']), 'video_url')
```

Comment:
Neither of the tests use this code. Can you give an example?

Reply:
Sorry, i don't get it, the `video_url` is used for `extract_m3u8_formats_and_subtitle`.

Reply:
Should i change to something like `manifest_url` ?

Reply:
sorry, the comment somehow ended up on the wrong line of code. I meant the `.replace` in L38 is not needed for either tests
```py
str(video_url).replace(r'\u002F', '/')
```

Reply:
What I mean is, this works for both tests
```py
        video_url = unquote(nuxtjs_data['video'])
        formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_url, display_id)
```

Reply:
I still get error for the second test after applying your suggestion

Log:
```shell
> python test/test_download.py TestDownload.test_TrtCocukVideo_1  
[debug] Loaded 1674 extractors 
[debug] [TrtCocukVideo] Extracting URL: https://www.trtcocuk.net.tr/video/sef-rokanin-lezzet-dunyasi-17 
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading webpage 
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading m3u8 information 
ERROR: [TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Failed to download m3u8 information: <urlopen error unknown url type: %22https> (caused by URLError('unknown url type: %22https'))
  File "[REDACTED]\extractor\common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "[REDACTED]\extractor\trtcocuk.py", line 34, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(
  File "[REDACTED]\extractor\common.py", line 2167, in _extract_m3u8_formats_and_subtitles
    res = self._download_webpage_handle(
  File "[REDACTED]\extractor\common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "[REDACTED]\extractor\common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "[REDACTED]\extractor\common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "[REDACTED]\YoutubeDL.py", line 3680, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "[REDACTED]\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "[REDACTED]\lib\urllib\request.py", line 541, in _open
    return self._call_chain(self.handle_open, 'unknown',
  File "[REDACTED]\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "[REDACTED]\lib\urllib\request.py", line 1419, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: %22https>

Retrying: 1 failed tries

##########


[debug] [TrtCocukVideo] Extracting URL: https://www.trtcocuk.net.tr/video/sef-rokanin-lezzet-dunyasi-17
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading webpage
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading m3u8 information 
ERROR: [TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Failed to download m3u8 information: <urlopen error unknown url type: %22https> (caused by URLError('unknown url type: %22https'))
  File "[REDACTED]\extractor\common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "[REDACTED]\extractor\trtcocuk.py", line 34, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(
  File "[REDACTED]\extractor\common.py", line 2167, in _extract_m3u8_formats_and_subtitles
    res = self._download_webpage_handle(
  File "[REDACTED]\extractor\common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "[REDACTED]\extractor\common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "[REDACTED]\extractor\common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "[REDACTED]\YoutubeDL.py", line 3680, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "[REDACTED]\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "[REDACTED]\lib\urllib\request.py", line 541, in _open
    return self._call_chain(self.handle_open, 'unknown',
  File "[REDACTED]\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "[REDACTED]\lib\urllib\request.py", line 1419, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: %22https>

Retrying: 2 failed tries

##########


[debug] [TrtCocukVideo] Extracting URL: https://www.trtcocuk.net.tr/video/sef-rokanin-lezzet-dunyasi-17
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading webpage
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading m3u8 information 
ERROR: [TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Failed to download m3u8 information: <urlopen error unknown url type: %22https> (caused by URLError('unknown url type: %22https'))
  File "[REDACTED]\extractor\common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "[REDACTED]\extractor\trtcocuk.py", line 34, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(
  File "[REDACTED]\extractor\common.py", line 2167, in _extract_m3u8_formats_and_subtitles
    res = self._download_webpage_handle(
  File "[REDACTED]\extractor\common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "[REDACTED]\extractor\common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "[REDACTED]\extractor\common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "[REDACTED]\YoutubeDL.py", line 3680, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "[REDACTED]\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "[REDACTED]\lib\urllib\request.py", line 541, in _open
    return self._call_chain(self.handle_open, 'unknown',
  File "[REDACTED]\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "[REDACTED]\lib\urllib\request.py", line 1419, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: %22https>

WARNING: test_TrtCocukVideo_1 failed due to network errors, skipping...
.
----------------------------------------------------------------------
Ran 1 test in 6.902s
```

Reply:
oh! Since the test got skipped, pytest was listing it as passed 😓